src/aggregated_non_cooperative.jl

```diff
@@ -189,7 +191,7 @@ function finalize_results!(::AbstractGroupANC, ECModel::AbstractEC)
     time_set = 1:n_steps
 
     project_lifetime = field(gen_data, "project_lifetime")
-    peak_categories = profile(market_data, "peak_categories")
+    peak_categories = market_profile_by_user(ECModel,u_standard,"peak_categories")
```

Comment:
I don't like this as you assume that all market types would share the same peak_categories.

What you could do is to define peak_categories and consequently also peak_set as dictionaries with:
``` julia
peak_caterogies = Dict(
   u=>arket_profile_by_user(ECModel,u_standard,"peak_categories")
   for u in user_set
  )
```

And then the peak_set can use this definition to apply unique to every element.

Then in the code below, where peak_set is used, you may just use peak_set[u].

My feeling is that there may be some weird side effects in the output plotting and so on, but let's check one by one

Reply:
I've tried to apply this changes but some variables such as:`    _P_max_agg = JuMP.Containers.DenseAxisArray(
        [maximum(abs.(ECModel.results[:P_agg][findall(==(w), peak_categories[u])].data))
        for w in peak_set[u]],
            peak_set)  # Maximum dispatch of the user for every peak period

    peak_aggregator = DataFrames.DataFrame(
        vcat(
            [[EC_CODE]],
            [[[_P_max_agg[w]]] for w in peak_set]...
        ),
        map(Symbol, ["Agg_id"; map(x->"Peak_id $x", peak_set)])
    )`

Seems that cannot accept peak_set[u] because it doesn't exist as single user. How do you think we could proceed?

Reply:
The problem is still present:
`    _P_max_agg = JuMP.Containers.DenseAxisArray(
        [maximum(abs.(ECModel.results[:P_agg][findall(==(w), peak_categories)].data))
        for w in peak_set],
            peak_set)  # Maximum dispatch of the user for every peak period

    peak_aggregator = DataFrames.DataFrame(
        vcat(
            [[EC_CODE]],
            [[[_P_max_agg[w]]] for w in peak_set]...
        ),
        map(Symbol, ["Agg_id"; map(x->"Peak_id $x", peak_set)])
    )`

It seems that when i create the output XLSX is not able to write it in the proper way

Reply:
We could add peak_categories also for the EC, this should bypass this issue.
Do you think you could implement this easily?